### PR TITLE
create the function gather_financial_statement_company_compare

### PR DESCRIPTION
**Pull request by sirine-chahma**
Created the function gather_financial_statement_company_compare and did the exception handling

### CODE REVIEW

**Comment by sirine-chahma** [comment]
Sorry, I pushed a lot of files, I have no idea what they are...

File: pyntrinio/pyntrinio.py [hunk 1]
```diff
@@ -110,56 +110,130 @@ def gather_financial_statement_company_compare(api_key, ticker, statement, year,
...
-  'EarningsPerShareDiluted': [0.77],
-  'WeightedAverageNumberOfSharesOutstandingBasic': [4565000000.0],
-  'WeightedAverageNumberOfDilutedSharesOutstanding': [4614000000.0]}}
-  """
+  """    
+  inputs = {'api_key':api_key, 'statement':statement, 'year':year, 'period':period}
+  #Check if api_key, statement, year, period are strings
+  for inst in inputs.keys():
+    if not isinstance(inputs[inst], str):
+      raise Exception("Sorry, " + inst + " must be a string")
```
Comment (slbsolomon) [inline pyntrinio/pyntrinio.py:118]: Very helpful to have specific issue in the error! Will consider adding this other places.

File: pyntrinio/pyntrinio.py [hunk 2]
```diff
@@ -110,56 +110,130 @@ def gather_financial_statement_company_compare(api_key, ticker, statement, year,
...
+          
+  #Check if ticker is a list
+  if not isinstance(ticker, list):
+    raise Exception("Sorry, ticker must be a list")
   
-  if output_format=='dict':
-    results={}
-  else:
-    results=pd.DataFrame(results)
+  #Check if the ticker is valid
```
Comment (slbsolomon) [inline pyntrinio/pyntrinio.py:124]: Is there an exception raised for this?

File: pyntrinio/pyntrinio.py [hunk 3]
```diff
@@ -110,56 +110,130 @@ def gather_financial_statement_company_compare(api_key, ticker, statement, year,
...
   
-  return results
+  #Check if the statement if valid
```
Comment (slbsolomon) [inline pyntrinio/pyntrinio.py:126]: Is there an exception raised for this? Can we add the 3 options into the docs?

File: pyntrinio/pyntrinio.py [hunk 4]
```diff
@@ -110,56 +110,130 @@ def gather_financial_statement_company_compare(api_key, ticker, statement, year,
...
+  #Check if the output_format is either 'dict' or 'pddf' 
+  if not output_format in ['dict', 'pddf']:
+    raise Exception("Sorry, output_format must be 'dict' or 'pddf'.")
+  
+  
+  #link with the API
+  intrinio_sdk.ApiClient().configuration.api_key['api_key'] = api_key
+  fundamentals_api = intrinio_sdk.FundamentalsApi()
+  
+  #result will contain a dictionnary for each company.
```
Comment (slbsolomon) [inline pyntrinio/pyntrinio.py:142]: *list, not dict - can we also change the type on the docs?

File: pyntrinio/pyntrinio.py [hunk 5]
```diff
@@ -110,56 +110,130 @@ def gather_financial_statement_company_compare(api_key, ticker, statement, year,
...
+    dict ={}
+    dict['ticker'] = comp
+    dict['statement'] = statement
+    dict['year'] = year
+    dict['period'] = period
+      
+    #we store all the values, balances, names and the tags
+    for i in range(len(my_fund)):
+      value = my_fund[i].value
+      tag_dic = my_fund[i].xbrl_tag
```
Comment (slbsolomon) [inline pyntrinio/pyntrinio.py:164]: great that this is dynamic!

**Review by slbsolomon — APPROVED** [review]
Function looks good and is as discussed - I added some inline comments with recommendations for docs, and can we add an exception for if the API key is invalid?

**Pull request merged**